pkg/common/config/config.go

```diff
@@ -735,3 +735,31 @@ func GetConfigPath(ctx context.Context) string {
 	}
 	return cfgPath
 }
+
+// GetSessionUserAgent returns clusterwise unique useragent
+func GetSessionUserAgent(ctx context.Context, useragent string) string {
+	log := logger.GetLogger(ctx)
+	if strings.TrimSpace(useragent) == "" {
+		useragent = "k8s-csi-useragent"
+	}
+	clusterFlavor, err := GetClusterFlavor(ctx)
+	if err != nil {
+		log.Errorf("failed retrieving cluster flavor. Error: %v", err)
+		return useragent
+	}
+	cfg, err := GetConfig(ctx)
+	if err != nil {
+		log.Errorf("failed to read config. Error: %+v", err)
+		return useragent
+	}
+	if clusterFlavor == cnstypes.CnsClusterFlavorVanilla {
+		if cfg.Global.ClusterID != "" {
```

Comment:
if cfg.Global.ClusterID is empty then we need to read auto generated cluster id from CR.
cc: @vdkotkar

Reply:
@vdkotkar should we read generated user agent from GeneratedVanillaClusterID field?
as:
```suggestion
		if cfg.Global.ClusterID != "" {
			useragent = useragent + "-" + cfg.Global.ClusterID
		} else {
			useragent = useragent + "-" + GeneratedVanillaClusterID
		}
```

Reply:
As discussed offline, you can't use GeneratedVanillaClusterID, as its scope is limited to vsphere controller container and not to syncer container. See getConfig wrapper function in syncer code which is added to fetch cluster ID.

Reply:
make sure we use generated Cluster ID only for vanilla and not for wcp.

Reply:
As discussed updated getConfig wrapper function,  where generated cluster-id will set to GeneratedVanillaClusterID so that it can be used in syncer container.  generated cluster-id will be used for vanilla, the check is added for that.

Reply:
I see cfg.Global.ClusterID is set when auto generated cluster id is fetched.
and we need to call this function after syncer and controller is done with reading config file while creating vc session.
so do not worry about generated cluster id and directly use  `cfg.Global.ClusterID`

Reply:
made changes as suggested

Reply:
after removing change related "GeneratedVanillaClusterID" , during testing found generated cluster-id is empty for syncer